diff -ur openjdk.orig/jdk/make/jdk_generic_profile.sh openjdk/jdk/make/jdk_generic_profile.sh
--- openjdk.orig/jdk/make/jdk_generic_profile.sh	2011-10-20 16:12:28.000000000 -0400
+++ openjdk/jdk/make/jdk_generic_profile.sh	2011-11-08 11:32:53.365858929 -0500
@@ -421,7 +421,7 @@
 # JPEG_CFLAGS and JPEG_LIBS tell the compiler how to compile and
 # link against libjpeg
 if [ "${JPEG_LIBS}" = "" ] ; then
-    ZLIB_LIBS="-ljpeg"
+    JPEG_LIBS="-ljpeg"
 fi
 export JPEG_LIBS
 
