diff -up ./openjdk/hotspot/make/linux/makefiles/vm.make.sav ./openjdk/hotspot/make/linux/makefiles/vm.make
--- ./openjdk/hotspot/make/linux/makefiles/vm.make.sav	2012-03-12 14:50:26.501783501 -0400
+++ ./openjdk/hotspot/make/linux/makefiles/vm.make	2012-03-12 15:02:27.884518939 -0400
@@ -87,6 +87,7 @@ HS_LIB_ARCH   = -DHOTSPOT_LIB_ARCH=\"$(L
 BUILD_TARGET  = -DHOTSPOT_BUILD_TARGET="\"$(TARGET)\""
 BUILD_USER    = -DHOTSPOT_BUILD_USER="\"$(HOTSPOT_BUILD_USER)\""
 VM_DISTRO     = -DHOTSPOT_VM_DISTRO="\"$(HOTSPOT_VM_DISTRO)\""
+REG_DIR       = -DREG_DIR="\"$(JNI_LOC)\""
 
 CPPFLAGS =           \
   ${SYSDEFS}         \
@@ -96,6 +97,7 @@ CPPFLAGS =           \
   ${BUILD_USER}      \
   ${HS_LIB_ARCH}     \
   ${JRE_VERSION}     \
+  ${REG_DIR}         \
   ${VM_DISTRO}
 
 ifdef DERIVATIVE_ID
diff -up ./openjdk/hotspot/src/os/linux/vm/os_linux.cpp.sav ./openjdk/hotspot/src/os/linux/vm/os_linux.cpp
--- ./openjdk/hotspot/src/os/linux/vm/os_linux.cpp.sav	2012-03-12 14:44:14.966559167 -0400
+++ ./openjdk/hotspot/src/os/linux/vm/os_linux.cpp	2012-03-12 14:44:33.485271287 -0400
@@ -393,7 +393,10 @@ void os::init_system_properties_values()
 
 #define EXTENSIONS_DIR  "/lib/ext"
 #define ENDORSED_DIR    "/lib/endorsed"
+
+#ifndef REG_DIR
 #define REG_DIR         "/usr/java/packages"
+#endif
 
   {
     /* sysclasspath, java_home, dll_dir */
