diff -up openjdk/jdk/make/common/shared/Defs-java.gmk.sav openjdk/jdk/make/common/shared/Defs-java.gmk
diff -up openjdk/jdk/make/com/sun/java/pack/Makefile.sav openjdk/jdk/make/com/sun/java/pack/Makefile
--- openjdk/jdk/make/com/sun/java/pack/Makefile.sav	2011-07-29 15:15:16.038605376 -0400
+++ openjdk/jdk/make/com/sun/java/pack/Makefile	2011-07-29 15:15:49.166082596 -0400
@@ -33,7 +33,6 @@ LIBRARY = unpack
 PRODUCT = sun
 PGRM = unpack200
 JAVAC_MAX_WARNINGS=true
-JAVAC_WARNINGS_FATAL=true
 include $(BUILDDIR)/common/Defs.gmk
 
 CPLUSPLUSLIBRARY=true
diff -up openjdk/jdk/make/tools/Makefile.sav openjdk/jdk/make/tools/Makefile
--- openjdk/jdk/make/tools/Makefile.sav	2011-07-29 15:14:45.728083910 -0400
+++ openjdk/jdk/make/tools/Makefile	2011-07-29 15:14:51.958985518 -0400
@@ -29,7 +29,7 @@
 
 BUILDDIR = ..
 
-SUBDIRS_MAKEFLAGS += JAVAC_MAX_WARNINGS=true JAVAC_WARNINGS_FATAL=true
+SUBDIRS_MAKEFLAGS += JAVAC_MAX_WARNINGS=true
 include $(BUILDDIR)/common/Defs.gmk
 
 # Note: freetypecheck is built by Sanity.gmk if needed
