diff --git a/make/javax/crypto/Makefile b/make/javax/crypto/Makefile
--- jdk8/jdk/make/javax/crypto/Makefile
+++ jdk8/jdk/make/javax/crypto/Makefile
@@ -158,7 +158,12 @@
 #
 
 ifdef OPENJDK
-all: build-jar install-jar build-policy install-limited
+ifdef UNLIMITED_CRYPTO
+POLICY = install-unlimited
+else
+POLICY = install-limited
+endif
+all: build-jar install-jar build-policy $(POLICY)
 else  # OPENJDK
 ifeq ($(strip $(FILES_java)),)
 all:
